server/controllers/categoryController.js

```diff
@@ -0,0 +1,360 @@
+        })
+          .then(() => (
+            res.status(202).json({
+              status: 'status',
+              category,
+            })
+          ))
+          .catch(error => (
+            res.status(400).json({
+              status: 'error',
+              error
+            })
+          ));
+      })
+      .catch(error => (
+        res.status(400).json({
+          status: 'error',
+          error
+        })
+      ));
+  }
+
+  /**
+   * @description Create a new category
+   * @param  {object} req body of the Admin's request
+   * @param  {function} res response from the server
+   * @returns {object} The body of the response message
+   * @memberof CategoryController
+   */
+  static deleteCategory(req, res) {
+    const { categoryName } = req.params;
+    Category.findOne({
+      where: { name: categoryName }
+    })
+      .then((category) => {
+        if (!category) {
+          return res.status(404).json({
+            status: 'error',
+            error: 'You cannot delete a ategory does not exist'
+          });
+        }
+        return category.destroy()
+          .then(() => (
+            res.status(204).json({
+              status: 'success',
+              error: 'Category deleted successfully'
+            })
+          ))
+          .catch(error => res.status(400).json({ status: 'error', error }));
+      })
+      .catch(error => (
+        res.status(400).json({
+          status: 'error',
+          error,
+        })
+      ));
+  }
+
+  /**
+   * @description Adds an article to a  category
+   * @param  {object} req body of the Author's request
+   * @param  {function} res response from the server
+   * @param  {function} next response from the server
+   * @returns {object} The body of the response message
+   * @memberof CategoryController
+   */
+  static addArticleToACategory(req, res, next) {
+    const articleTitle = req.body.articleTitle;
+    const categoryName = req.params.categoryName;
+    // Check if category exists
+    Category.findOne({
+      where: { name: categoryName }
+    })
+      .then((category) => {
+        if (!category) {
+          return res.status(404).json({
+            status: 'error',
+            error: 'Category does not exist'
+          });
+        }
+        // If category exists, check if article exists
+        Article.findOne(({
+          where: { title: articleTitle }
+        }))
+          .then((article) => {
+            if (!article) {
+              return res.status(404).json({
+                status: 'error',
+                error: 'Article does not exist'
+              });
+            }
+            const categoryId = category.id;
+            const articleId = article.id;
+            const { authorization } = req.headers;
+            const token = authorization.split(' ')[1];
+            const decoded = TokenHelper.decodeToken(token);
+            const userId = decoded.id;
+            if (article.authorId !== userId) {
```

Comment:
Your logic looks good here, talking about the `addArticleToACategory` method. However, I do not see the need for you to decode the token again here (lines `208-212`) since you have already called the `authenticateUser` and `authorizeAuthor` middlewares before this method is called.

You can simply access the current `userId` from the `req` object instead and use it.

Reply:
ok. I'll fix that

Reply:
Fixed